webapp/src/app.tsx

```diff
@@ -1911,7 +1911,7 @@ function isProjectFile(filename: string): boolean {
 }
 
 function isPNGFile(filename: string): boolean {
-    return /\.png$/i.test(filename)
+    return pxt.appTarget && pxt.appTarget.compile && pxt.appTarget.compile.saveAsPNG && /\.png$/i.test(filename);
```

Comment:
I think you just need `return pxt.appTarget.compile.saveAsPNG && /\.png$/i.test(filename);`

Reply:
You definitely need the compile check, but yeah you're right about checking for appTarget. I was a tad overzealous and forgot this is in the webapp 😄

Reply:
No you don't. https://github.com/Microsoft/pxt/blob/master/pxtlib/main.ts#L20

Reply:
I stand corrected! We definitely check it all over the place though.

Reply:
I know... wasting precious cycles...